lucene/codecs/src/java/org/apache/lucene/codecs/uniformsplit/sharedterms/UnionFieldMetadataBuilder.java

```diff
@@ -27,34 +27,48 @@
  */
 public class UnionFieldMetadataBuilder {
 
+  private long dictionaryStartFP;
```

Comment:
Store more metadata for determining the dictionary load mode the same way as regular FieldMetadata for a single field.